runtime/src/util/map.rs

```diff
@@ -0,0 +1,198 @@
+use crate::builtin::HAMT_BIT_WIDTH;
+use crate::{ActorError, AsActorError, Hasher};
+use anyhow::anyhow;
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_hamt as hamt;
+use fvm_shared::address::Address;
+use fvm_shared::error::ExitCode;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::marker::PhantomData;
+
+/// Wraps a HAMT to provide a convenient map API.
+/// Any errors are returned with exit code indicating illegal state.
+/// The name is not persisted in state, but adorns any error messages.
+pub struct Map2<'bs, BS, K, V>
+where
+    BS: Blockstore,
+    K: MapKey,
+    V: DeserializeOwned + Serialize,
+{
+    hamt: hamt::Hamt<&'bs BS, V, hamt::BytesKey, Hasher>,
```

Comment:
nit: I'd make this own the underlying blockstore. There's a blanket `Blockstore for &Blockstore` impl anyways.

Reply:
I.e., even if it owns the blockstore, you can still use `Map2<&SomeBS, ...>`.

Reply:
Done